src/gameplay_v2/systems/leaderSelectionS.ts

```diff
@@ -0,0 +1,76 @@
+import { GameData } from '../gameEngine';
+import { SendData, System } from './system';
+import { Leader } from '../roles/components/leader';
+import { Player } from '../player';
+
+export class LeaderSelectionS implements System {
+    run(gameData: GameData, sendData: SendData): Player[] {
+        
+        const noLeaderLists:Player[] = [];
+        const playerList:Player[] = gameData.players;
+        let newLeaderinPlayerList:Player[] = [];
+        let currentLeader:string;
+        let leaderPosition = 0;
+
+        for (const player of playerList) {
+            
+            // find the current leader
+            if (player.entity.components.find((component) => component.name === "leader") ) {
+                currentLeader = player.username;
+                break;
+            }
+            leaderPosition++;
+            noLeaderLists.push(player);
+
+        }
+        // if the leader is not set
+        if (currentLeader == null) {
+            playerList[0].entity.components.push(new Leader());
+
+            return playerList;
+        }
+
+        newLeaderinPlayerList = this.changeLeader(currentLeader, leaderPosition, playerList);
+   
+
+        return newLeaderinPlayerList;
+
+    }
+
+    // swith leader to the next player    
+    changeLeader(username: string, leaderPosition:number, playerList: Player[]): Player[] {
```

Comment:
We can simply call this "incrementLeader". We should never be jumping leaders. Although, I guess a cool role or effect would be to hard set a new leader. 

For now, let's pin it down to just incrementing a leader so it's less easy to abuse and misuse. We can extend it back out to this again later if we need to!